srcs/app/http/HttpRequestHandler.cpp

```diff
@@ -22,92 +29,292 @@ bool		HttpRequestHandler::GetKeepAlive() const {
+	if (location) {
+		return location->common;
+	}
+	return server_config_.common;
+}
+
+bool	HttpRequestHandler::HasAcceptedFormat_(const Location *location,
+												const HttpRequest &request) {
+	if (location && !location->limit_except.empty()) {
+		if (std::find(location->limit_except.begin(),
+					location->limit_except.end(),
+					request.GetMethod()) == location->limit_except.end()) {
+			RequestError_(location, 405);
+			return false;
+		}
+	}
+	const CommonConfig &cfg = GetCommonConfig(location);
+	if (request.GetBody().size() > cfg.client_max_body_size) {
+		RequestError_(location, 413);
+		return false;
+	}
+	if (request.HasHeader("Content-Encoding")) {
+		RequestError_(location, 415);
+		return false;
+	}
+	return true;
+}
+
+void	HttpRequestHandler::ServeFile_(const Location *location,
+												const std::string &file_path) {
+	if (!IsRegularFile_(file_path)) {
+		RequestError_(location, 403);
+		return;
+	}
+	std::ifstream ifs(file_path.c_str(), std::ios::in | std::ios::binary);
+	if (!ifs) {
+		RequestError_(location, 404);
+		return;
+	}
+	const std::string body = std::string(std::istreambuf_iterator<char>(ifs),
+											std::istreambuf_iterator<char>());
+	HttpResponse response(200);
+	AddCommonHeaders_(&response);
+	response.AddHeader("Content-Type", GetMimeType_(file_path));
 	response.SetBody(body);
+	raw_response_ = response.CreateResponseString();
+}
+
+void	HttpRequestHandler::MovedPermanently_(const HttpRequest &request) {
+	HttpResponse response(301);
 	AddCommonHeaders_(&response);
+	response.AddHeader("Content-Type", "text/html");
+	std::stringstream url;
+	url << "http://" << request.GetHost() << ":" << request.GetPort() <<
+		request.GetPath() << "/";
+	response.AddHeader("Location", url.str());
+	const std::string body = DefaultResponseBody_(301);
+	response.SetBody(body);
 	raw_response_ = response.CreateResponseString();
 }
 
-void	HttpRequestHandler::DoPost_(const HttpRequest &request) {
+void	HttpRequestHandler::DoGet_(const Location *location,
+									const HttpRequest &request) {
+	if (location && !location->common.return_url.empty()) {
+		DoRedirection_(location);
+		return;
+	}
+	if (!HasAcceptedFormat_(location, request)) {
+		return;
+	}
+	const std::string full_path = GetFullPath_(location, request.GetPath());
+	if (!IsValidPath_(full_path)) {
+		PathError_(location);
+		return;
+	}
+	if (IsRegularFile_(full_path)) {
+		ServeFile_(location, full_path);
+	} else {
+		const CommonConfig &cfg = GetCommonConfig(location);
+		const bool has_end_slash = full_path[full_path.size() - 1] == '/';
+		if (!has_end_slash) {
+			MovedPermanently_(request);
+			return;
+		}
+		const std::string index_path = full_path + cfg.index;
+		if (!cfg.autoindex || IsRegularFile_(index_path)) {
+			ServeFile_(location, index_path);
+		} else {
+			ListDirectory_(location, request.GetPath());
+		}
+	}
+}
+
+void	HttpRequestHandler::DoPost_(const Location *location,
```

Comment:
I have seen that the linter demands that we do not use references if not const, but rather pointers. Why is the opposite not true? Why not demand to use reference instead of pointers, when they are to const data?
I'm not asking you to change your implementation, I genuinelly don't know what is best, and why did you chose const Location * type instead of const Location &.

Reply:
I'm not sure why the linter does not demand a const reference instead of a const pointer.

In my opinion, it makes sense for pointers instead of references with non-const data as a form of documentation.

And, yes, in this case, we use Location as both a boolean and a way to get the information we need, so a pointer made the most sense to me, but there may be a better way to do it.